src/hooks/useENSResolveName.ts

```diff
@@ -4,14 +4,38 @@ import {
   saveResolveName,
 } from '@rainbow-me/handlers/localstorage/ens';
 import { web3Provider } from '@rainbow-me/handlers/web3';
+import { queryClient } from '@rainbow-me/react-query/queryClient';
 
-export default function useENSResolveName(ensName: string) {
-  return useQuery(['resolve-name', ensName], async () => {
-    const cachedAddress = await getResolveName(ensName);
-    if (cachedAddress) return cachedAddress;
+const STALE_TIME = 10000;
+
+export const ensResolveNameQueryKey = (ensName: string) => [
+  'resolve-name',
+  ensName,
+];
+
+async function resolveENSName(ensName: string) {
+  const cachedAddress = await getResolveName(ensName);
+  if (cachedAddress)
+    queryClient.setQueryData(ensResolveNameQueryKey(ensName), cachedAddress);
+  const address = await web3Provider.resolveName(ensName);
```

Comment:
can we use `fetchPrimary` from handlers ens instead? that one uses `getProviderForNetwork` instead of `web3Provider`

Reply:
thinking out loud, maybe `fetchPrimary` is not the best name for that method